File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        head = None
        temp = None

        if (list1 == None):
            return list2
        if (list2 == None):
            return list1

        if (list1.val <= list2.val):
            head = list1
            temp = list1
            list1 = list1.next
        else:
            head = list2
            temp = list2
            list2 = list2.next
        
        while(list1 != None and list2 != None):
            if (list1.val <= list2.val):
                temp.next = list1
                list1 = list1.next
            else:
                temp.next = list2
                list2 = list2.next
            temp = temp.next

        if (list1 == None):
            temp.next = list2
        if (list2 == None):
            temp.next = list1
        return head

        

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        if (lists.length > 0) {
            ListNode head = null;
            ListNode temp = null;
            for (int j = 0; j != lists.length; j = 0) {
                int minIndeks = -1;
                for (int i = 0; i < lists.length; i++) {
                    if (lists[i] != null) {
                        if (minIndeks == -1 || lists[i].val < lists[minIndeks].val) {
                            minIndeks = i;
                        }
                    } else {
                        j++;
                    }
                }
                if (j == lists.length) {
                    break;
                }
                if (head == null) {
                    head = new ListNode(lists[minIndeks].val);
                    temp = head;
                } else {
                    ListNode newNode = new ListNode(lists[minIndeks].val);
                    temp.next = newNode;
                    temp = temp.next;
                }
                lists[minIndeks] = lists[minIndeks].next;
            }
            return head;
        }
        return null;
    }
}
